aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -1,442 +1,59 @@
-            if self
-                .idx_to_update
-                .contains_key(&ShiftedTxnIndex::zero_idx())
-            {
-                MVGroupError::TagNotFound
-            } else {
-                MVGroupError::Uninitialized
-            }
-        };
-
-        self.versioned_map
-            .get(tag)
-            .ok_or(common_error())
-            .and_then(|tree| {
-                match tree
-                    .range(ShiftedTxnIndex::zero_idx()..ShiftedTxnIndex::new(txn_idx))
-                    .next_back()
-                {
-                    Some((idx, entry)) => {
-                        if entry.flag == Flag::Estimate {
-                            Err(MVGroupError::Dependency(
-                                idx.idx()
-                                    .expect("Base version cannot be marked as estimate"),
-                            ))
-                        } else {
-                            Ok((
-                                idx.idx().map(|idx| (idx, entry.incarnation)),
-                                entry.value.clone(),
-                            ))
-                        }
-                    },
-                    None => Err(common_error()),
-                }
-            })
-    }
-
-    fn get_latest_group_size(
-        &self,
-        shifted_txn_idx: &ShiftedTxnIndex,
-    ) -> Result<ResourceGroupSize, MVGroupError> {
-        self.idx_to_update
-            .range(ShiftedTxnIndex::zero_idx()..shifted_txn_idx.clone())
-            .next_back()
-            .map(|(idx, group_update)| {
-                // We would like to use the value in an estimated entry if size never changed
-                // between speculative executions, i.e. to depend on estimates only when the
-                // size has changed. In this case, execution can wait on a dependency, while
-                // validation can short circuit to fail.
-                if group_update.flag == Flag::Estimate && self.size_changed {
-                    Err(MVGroupError::Dependency(
-                        idx.idx().expect("May not depend on storage version"),
-                    ))
-                } else {
-                    Ok(group_update.size)
-                }
-            })
-            .unwrap_or(Err(MVGroupError::Uninitialized))
-    }
+    // TODO: Optimize the key represetantion to avoid cloning and concatenation for APIs
+    // such as get, where only & of the key is needed.
+    values: VersionedData<(K, T), V>,
+    // TODO: Once AggregatorV1 is deprecated (no V: TransactionWrite trait bound),
+    // switch to VersionedVersionedData<K, ResourceGroupSize>.
+    // If an entry exists for a group key in Dashmap, the group is considered initialized.
+    group_sizes: DashMap<K, GroupSizes>,
+
+    // Stored separately from group sizes to avoid R-W lock contention between
+    // latest group contents preparation and the size updates.
+    group_tags: DashMap<K, HashSet<T>>,
```

Comment:
can you add a comment on what value this should have, based on changes that can happen?